dev-lang/rust_bin/rust_bin-1.29.0.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="Modern and safe systems programming language"
+DESCRIPTION="Rust is a systems programming language that runs blazingly fast, \
+prevents almost all crashes*, and eliminates data races."
+HOMEPAGE="https://www.rust-lang.org/"
+COPYRIGHT="2018 The Rust Project Developers"
+LICENSE="MIT"
+REVISION="1"
+
+case "$effectiveTargetArchitecture" in
+	x86)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-i686-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="b4d40d82bb6c07746b1ca8e407a23e45dd357744ee80217446bc9f25cc5e9eaa"
+SOURCE_DIR="rust-$portVersion-i686-unknown-haiku"
+		;;
+	x86_64)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-x86_64-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="e8229ee38453978485720e6a8ef84e0474aa5fb9375b3e74ecce655ee742b5f2"
+SOURCE_DIR="rust-$portVersion-x86_64-unknown-haiku"
+		;;
+	*)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rustc-$portVersion-src.tar.xz"
+CHECKSUM_SHA256="cec7232a3081f7a29919523490b7045dd9637aee371de4051c4f9348b657f30e"
+SOURCE_DIR="rustc-$portVersion-src"
+		;;
+esac
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+DISABLE_SOURCE_PACKAGE=yes
+
+cargoVersion="0.30.0"
+rlsVersion="0.130.0"
+rustfmtVersion="0.99.1"
+clippyVersion="0.0.212"
+
+PROVIDES="
+	rust_bin$secondaryArchSuffix = $portVersion
+	cmd:rustc = $portVersion
+	cmd:rustdoc = $portVersion
+	cmd:rust_gdb = $portVersion
+	cmd:rust_lldb = $portVersion
+	cmd:cargo$secondaryArchSuffix = $cargoVersion
+	cmd:cargo_clippy = $clippyVersion
+	cmd:cargo_fmt = $cargoVersion
+	cmd:clippy_driver = $clippyVersion
+	cmd:rls = $rlsVersion
+	cmd:rustfmt = $rustfmtVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libssh2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+CONFLICTS="
+	rust$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+INSTALL()
+{
+	./install.sh                                   \
+		--prefix=$prefix                           \
+		--docdir=$developDocDir                    \
+		--libdir=$libDir                           \
+		--mandir=$manDir                           \
+		--sysconfdir=$dataDir                      \
+		--disable-ldconfig
+
+	# move the cargo and binaries (in case of a secondary arch)
+	if [ -n "$secondaryArchSuffix" ]; then
+		mkdir -p $binDir
+		mv $prefix/bin/cargo $binDir/cargo
+	fi
```

Comment:
Maybe:
```diff
-	# move the cargo and binaries (in case of a secondary arch)
-	if [ -n "$secondaryArchSuffix" ]; then
-		mkdir -p $binDir
-		mv $prefix/bin/cargo $binDir/cargo
-	fi
-
```